docs/source/concepts/lifecycle.rst

```diff
@@ -1,22 +1,148 @@
+.. _lifecycle_concept:
+
+========================
 The Simulation Lifecycle
 ========================
 
+A ``vivarium`` simulation has many important lifecycle phases.
+
+It begins with a user interacting with one of the simulation
+:ref:`entry points <entry_points_concept>`. The simulation then goes through an
+initialization cycle that creates all the :term:`components <Component>`,
+and compiles :term:`configuration information <Configuration>`.  After
+initialization, the simulation enters the setup phase, where each component
+in turn has a special ``setup`` method called with the simulation
+:ref:`builder <builder_concept>`.  During the setup, components ask for
+services from the framework and register themselves with various simulation
+subsystems.  There is then a post setup phase where the simulation framework
+does some cleanup.
+
+The simulation then enters its main processing section. The population
+is initialized, then the time step loop begins. Once the loop completes,
+results are calculated, gathered, and written out.
+
 .. contents::
    :depth: 2
    :local:
    :backlinks: none
 
-Outline
--------
-
-- Entry point variations (link to more detailed guide)
-- Initialization (create objects, compile configuration)
-- Setup (Wire components, load data)
-- Post setup
-- Population Initialization
-- Event loop
-  - time_step__prepare
-  - time_step
-  - time_step__cleanup
-  - collect_metrics
-- simulation_end
+Initialization
+--------------
+
+The initialization phase of ``vivarium`` lasts from when a user first interacts
+with a simulation :ref:`entry point <entry_points_concept>` to when the
+``__init__`` method of the :class:`vivarium.framework.engine.SimulationContext`
+completes.  In it, the following things happen (roughly in this order):
+
+1. Simulation outputs are configured. This is usually just setting up some
+   directories for the final results and the simulation logs.
+2. :term:`Model specification <Model Specification>` defaults are collected and
+   compiled along with the model specification file (if provided) into a
+   :class:`vivarium.config_tree.ConfigTree` object.
+3. A :class:`vivarium.framework.plugins.PluginManager` is generated around the
+   :term:`plugins <Plugin>` section of of the model specification.  The plugin
+   manager is responsible for parsing the plugin section and instantiating
+   plugin controllers and interfaces for the framework.
+4. The :class:`vivarium.framework.components.ComponentConfigurationParser` is
+   created to parse and instantiate all standard simulation
+   :term:`components <Component>`, if not provided directly by the user in
```

Comment:
this is a little confusing - is this saying if the components aren't provided directly by the user or if the ComponentConfigurationParser is not provided by the user?